common/fixation_entry.go

```diff
@@ -169,38 +171,50 @@ func handleRefAction(ctx sdk.Context, entry *types.Entry, refAction types.Refere
 	case types.SUB_REFERENCE:
 		if entry.GetRefcount() > 0 {
 			entry.Refcount -= 1
+		} else {
+			return utils.LavaError(ctx, ctx.Logger(), "handleRefAction_sub_ref_from_non_positive_count", map[string]string{"refCount": strconv.FormatUint(entry.GetRefcount(), 10)}, "refCount is not larger than zero. Can't subtract refcount")
 		}
 	case types.DO_NOTHING:
 	}
 
+	// get the relevant store
+	store := prefix.NewStore(ctx.KVStore(fs.storeKey), types.KeyPrefix(fs.createStoreKey(entry.GetIndex())))
```

Comment:
why is this here? we already have the store in the function above, this func should only handle the refCount, not saving the entry

Reply:
moved it to the function above